Flaky test SharedProcessorTest.testDistributeAfterStop

```
2026-05-04T13:36:29.4452196Z [ERROR] Tests run: 2, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 0.046 s <<< FAILURE! -- in io.fabric8.kubernetes.client.informers.impl.cache.SharedProcessorTest
2026-05-04T13:36:29.4568432Z [ERROR] io.fabric8.kubernetes.client.informers.impl.cache.SharedProcessorTest.testDistributeAfterStop -- Time elapsed: 0.017 s <<< ERROR!
2026-05-04T13:36:29.4621344Z java.lang.NullPointerException
2026-05-04T13:36:29.4642859Z 	at io.fabric8.kubernetes.client.utils.internal.SerialExecutor.execute(SerialExecutor.java:57)
2026-05-04T13:36:29.4683842Z 	at io.fabric8.kubernetes.client.informers.impl.cache.SharedProcessor.distribute(SharedProcessor.java:110)
2026-05-04T13:36:29.4741457Z 	at io.fabric8.kubernetes.client.informers.impl.cache.SharedProcessor.distribute(SharedProcessor.java:91)
2026-05-04T13:36:29.4793692Z 	at io.fabric8.kubernetes.client.informers.impl.cache.SharedProcessorTest.testDistributeAfterStop(SharedProcessorTest.java:62)
2026-05-04T13:36:29.4826871Z 	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
2026-05-04T13:36:29.4911356Z 	at java.base/java.util.ArrayList.forEach(ArrayList.java:1541)
2026-05-04T13:36:29.4949363Z 	at java.base/java.util.ArrayList.forEach(ArrayList.java:1541)
```

--- a/kubernetes-client/src/main/java/io/fabric8/kubernetes/client/utils/internal/SerialExecutor.java
+++ b/kubernetes-client/src/main/java/io/fabric8/kubernetes/client/utils/internal/SerialExecutor.java
@@ -55,6 +55,7 @@ public class SerialExecutor implements Executor {
   public synchronized void execute(final Runnable r) {
     if (shutdown) {
       logger.debug("Task submitted after the executor was shutdown");
+      return;
     }
     tasks.offer(() -> {
       try {

--- a/kubernetes-client/src/test/java/io/fabric8/kubernetes/client/informers/impl/cache/SharedProcessorTest.java
+++ b/kubernetes-client/src/test/java/io/fabric8/kubernetes/client/informers/impl/cache/SharedProcessorTest.java
@@ -20,6 +20,7 @@ import io.fabric8.kubernetes.api.model.PodBuilder;
 import io.fabric8.kubernetes.client.informers.ResourceEventHandler;
 import org.junit.jupiter.api.Test;
 
+import static org.junit.jupiter.api.Assertions.assertDoesNotThrow;
 import static org.junit.jupiter.api.Assertions.assertTrue;
 
 class SharedProcessorTest {
@@ -59,7 +60,7 @@ class SharedProcessorTest {
     ProcessorListener.Notification<Pod> addNotification = new ProcessorListener.AddNotification<>(foo1);
 
     // nothing should happen
-    sharedProcessor.distribute(addNotification, false);
+    assertDoesNotThrow(() -> sharedProcessor.distribute(addNotification, false));
   }
 
   private static class ExpectingNotificationHandler<T> extends ProcessorListener<T> {

--- a/kubernetes-client/src/test/java/io/fabric8/kubernetes/client/utils/internal/SerialExecutorTest.java
+++ b/kubernetes-client/src/test/java/io/fabric8/kubernetes/client/utils/internal/SerialExecutorTest.java
@@ -22,6 +22,7 @@ import java.util.concurrent.CountDownLatch;
 import java.util.concurrent.ExecutorService;
 import java.util.concurrent.Executors;
 import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicInteger;
 
 import static org.assertj.core.api.Assertions.assertThat;
 import static org.junit.jupiter.api.Assertions.assertFalse;
@@ -53,6 +54,20 @@ class SerialExecutorTest {
     Thread t = future.join();
     t.interrupt(); // interrupt the first task
     assertFalse(interrupted.join()); // make sure the second is not interrrupted
+  }
+
+  @Test
+  void executeAfterShutdownDoesNothing() {
+    final AtomicInteger underlyingExecutions = new AtomicInteger();
+    final AtomicInteger taskRuns = new AtomicInteger();
+    final SerialExecutor se = new SerialExecutor(r -> {
+      underlyingExecutions.incrementAndGet();
+      r.run();
+    });
+    se.shutdownNow();
+    se.execute(taskRuns::incrementAndGet);
+    assertThat(taskRuns).hasValue(0);
+    assertThat(underlyingExecutions).hasValue(0);
   }
 
   @Test

